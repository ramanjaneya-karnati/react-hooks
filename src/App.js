import React, { useState, useEffect } from 'react';
import './App.css';

export default function App() {
  const [count, setCount] = useState(0);
  useEffect(()=>{
    document.title=`pressed these ${count} time`
  });
  return(
    <div>
      <p>You clicked on {count} times</p>
      <button onClick={() => setCount(count + 1)}>Press Me</button>
    </div>
  )
}
