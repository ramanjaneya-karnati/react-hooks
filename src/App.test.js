import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import sinon from 'sinon';

describe('Rendering the component', ()=>{

  const wrapper = shallow(<App/>);

  it('Renders the component without crash', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should simulate the button clicks', function () {
    const buttonClick = sinon.spy();
    const wrapper = shallow(<App onClick={buttonClick}/>);
    wrapper.find('button').simulate('click');
    expect(buttonClick).toHaveProperty('callCount', 0);
  });
});

