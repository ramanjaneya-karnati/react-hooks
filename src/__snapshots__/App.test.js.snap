// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the component Renders the component without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          You clicked on 
          0
           times
        </p>,
        <button
          onClick={[Function]}
        >
          Press Me
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You clicked on ",
            0,
            " times",
          ],
        },
        "ref": null,
        "rendered": Array [
          "You clicked on ",
          0,
          " times",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Press Me",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Press Me",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            You clicked on 
            0
             times
          </p>,
          <button
            onClick={[Function]}
          >
            Press Me
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You clicked on ",
              0,
              " times",
            ],
          },
          "ref": null,
          "rendered": Array [
            "You clicked on ",
            0,
            " times",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Press Me",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Press Me",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Renders the counter component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          You clicked on 
          0
           times
        </p>,
        <button
          onClick={[Function]}
        >
          Press Me
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You clicked on ",
            0,
            " times",
          ],
        },
        "ref": null,
        "rendered": Array [
          "You clicked on ",
          0,
          " times",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Press Me",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Press Me",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            You clicked on 
            0
             times
          </p>,
          <button
            onClick={[Function]}
          >
            Press Me
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You clicked on ",
              0,
              " times",
            ],
          },
          "ref": null,
          "rendered": Array [
            "You clicked on ",
            0,
            " times",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Press Me",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Press Me",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
